Integrations/Digital_Defense_FrontlineVM/Digital_Defense_FrontlineVM.py

```diff
@@ -0,0 +1,795 @@
+        timeout_time = get_function_timeout_time(data_count)
+        signal.signal(signal.SIGALRM, function_timeout)
+        signal.alarm(timeout_time)
+        try:
+            all_data = get_all_data(current_data)
+            data.extend(all_data)
+        except EndOfTime:
+            return_error("Error: FrontlineVM get_fvm_data function exceeds timeout time.")
+        except Exception as err:
+            return_error("Error: FrontlineVM get_fvm_data failed. \n" + str(err))
+    return data
+
+
+def parse_params(param_dict):
+    '''
+        This parses the given dictionary and modifies it to comply with our API endpoint queries
+        of indexing multiple queries (?_0_first_query=value0_1_query1=value_2_query2=value2)
+    '''
+    param_index = 0
+    new_param_dict = {}
+    for key in param_dict:
+        new_key = "_" + str(param_index) + "_" + key
+        new_param_dict[new_key] = param_dict[key]
+        param_index += 1
+    return new_param_dict
+
+
+def get_query_date_param(day_input):
+    ''' Returns a datetime object of days from now to given day_input   '''
+    now = datetime.utcnow()
+    query_date = (now - timedelta(days=int(day_input))).replace(hour=0, minute=0, second=0, microsecond=0)
+    query_date = datetime.strftime(query_date, "%Y-%m-%dT%H:%M:%SZ")
+    return query_date
+
+
+def get_fetch_frequency_td():
+    ''' Returns the INCIDENT_FREQUENCY as a datetime.timedelta object    '''
+    fetch_frequency = INCIDENT_FREQUENCY.split()
+    demisto.debug("FrontlineVM get_fetch_incident_td -- using frequency: " + str(fetch_frequency))
+    if "min" in str(fetch_frequency[1]):
+        return timedelta(minutes=int(fetch_frequency[0]))
+    return timedelta(hours=int(fetch_frequency[0]))
+
+
+def create_vuln_event_object(vuln):
+    ''' Creates a vulnerability event object given raw vulnerability data.  '''
+    vuln_event = {}
+    vuln_event['vuln-id'] = vuln.get('id')
+    vuln_event['hostname'] = vuln.get('hostname')
+    vuln_event['ip-address'] = vuln.get('ip_address')
+    vuln_event['port'] = vuln.get("port")
+    vuln_event['scan-id'] = vuln.get('scan_id')
+    vuln_event['vuln-title'] = vuln.get('title')
+    vuln_event['date-created'] = vuln.get('active_view_date_created')
+    vuln_event['ddi-severity'] = vuln['severities']['ddi']
+    vuln_event['vuln-info'] = vuln.get('data')
+    return vuln_event
+
+
+def vulns_to_incident(vulns, start_time_dt):
+    '''
+        Iterate through vulnerabilities and create incident if
+        vulnerability has been created since last start_time.
+    '''
+    incidents = []
+    for vuln in vulns:
+        # get vulnerability active view (av) date created values:
+        av_date_created_str = vuln.get('active_view_date_created')
+        av_date_created_dt = datetime.strptime(av_date_created_str, "%Y-%m-%dT%H:%M:%S.%fZ")
+
+        # Skip if vuln created before current start_time:
+        if av_date_created_dt < start_time_dt:
```

Comment:
how could this happen if you sent the start_time in the request?

Reply:
Although we use the start_time to pull vulnerabilities, only vulnerabilities matching greater than or equal to that specific date will be pulled. 
To compare a more precise time, such as minutes, we would have to compare manually here.